src/trainer.py

```diff
@@ -509,25 +511,14 @@ def train(
 
         # define these here b/c they might get overridden on load
         n_pass, should_stop = 0, False
-        if self._serialization_dir is not None and phase == "pretrain":
+        if self._serialization_dir is not None:
             # Resume from serialization path
-            if load_model and any(
```

Comment:
Where does this logic go?

Reply:
The logic of searching for model_state checkpoints in the run directory is moved to its own function (check_for_previous_checkpiont in src/utils/utils.py). This part also becomes phase-agnostic as opposed to only for pretrain.